src/components/FilterableProductTable.js

```diff
@@ -0,0 +1,44 @@
+import React, { Component } from 'react'
+import ProductTable from './ProductTable'
+import SearchBar from './SearchBar'
+import '../App.css'
+
+export class FilterableProductTable extends Component {
+
+    state = {
+        products: this.props.products.data,
+    }
+
+    handleSearch = (value) => {
+        const newProducts = this.state.products.filter(item => (item.name.toUpperCase().includes(value.toUpperCase())))
+        
+        let showProducts = newProducts > 0 ? newProducts : this.state.products
+        // console.log(`NEW PRODUCTS: ${newProducts}`)
+        // console.log(`SHOW PRODUCTS: ${showProducts}`)
+
+        this.setState({
+            products: newProducts
+        })
+      }
+
+    showInStock = (value) => {
+        const productsInStock = value ? this.props.products : this.state.products.filter(product => product.stocked === true);
```

Comment:
you are missing the the data in the true statement:

```
showInStock = (value) => {
    const productsInStock = value
      ? this.props.products.data
      : this.state.products.filter((product) => product.stocked === true);

    this.setState({
      products: productsInStock,
    });
  };
```
That's why you were having an error! 

Good job overall mate!